File: src/main/java/uk/co/threebugs/mochiwhattotrade3/metatrader/TradeInfo.java
package uk.co.threebugs.mochiwhattotrade3.metatrader;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Value
@Jacksonized
@Builder
public class TradeInfo {
        int magic;
        BigDecimal lots;
        String symbol;
        BigDecimal swap;

        @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy.MM.dd HH:mm:ss")
        @JsonProperty("open_time")
        LocalDateTime openTime;

        @JsonProperty("SL")
        BigDecimal stopLoss;

        String comment;
        String type;

        @JsonProperty("open_price")
        BigDecimal openPrice;

        @JsonProperty("TP")
        BigDecimal takeProfit;

        @JsonProperty("pnl")
        BigDecimal profitAndLoss;

        Object mapType;

        Boolean empty;
}

File: src/main/kotlin/uk/co/threebugs/darwinexclient/metatrader/TradeInfo.kt
package uk.co.threebugs.darwinexclient.metatrader

import com.fasterxml.jackson.annotation.JsonFormat
import com.fasterxml.jackson.annotation.JsonProperty
import java.math.BigDecimal
import java.time.LocalDateTime


data class TradeInfo (
    var magic: Int? = null,
    var lots: BigDecimal? = null,
    var symbol: String? = null,
    var swap: BigDecimal? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy.MM.dd HH:mm:ss")
    @JsonProperty("open_time")
    var openTime: LocalDateTime? = null,

    @JsonProperty("SL")
    var stopLoss: BigDecimal? = null,
    var comment: String? = null,
    var type: String? = null,

    @JsonProperty("open_price")
    var openPrice: BigDecimal? = null,

    @JsonProperty("TP")
    var takeProfit: BigDecimal? = null,

    @JsonProperty("pnl")
    var profitAndLoss: BigDecimal? = null,
    var mapType: Any? = null,
    var empty: Boolean? = null
)

